src/v/io/tests/interval_map_test.cc

```diff
@@ -287,7 +287,7 @@ TEST(IntervalMap, RandomIntervals) {
 
         // helper to find the value index for a target offset
         auto value_index = [&](uint64_t target) -> std::optional<int> {
```

Comment:
> `-> std::optional<int>`

nitpick: Now there's an implicit narrowing conversion to signed

Reply:
i'm not sure i follow. like this patch creates a narrowing case or the patch shouldn't be necessary?

Reply:
ahh: you mean `return i;` -- yeh, fair. i think here in tests it's ok. it's clear the value of these types are going to be small, just placating the compiler!

Reply:
> i think here in tests it's ok

oh yeah i agree. just thought it might move the red line from the comparison the the return statement